### PR TITLE
Use GraalVM CE 22.3.1 to fix Nightly Build

### PR DESCRIPTION
**Pull request by linghengqian**
For #21347.

Changes proposed in this pull request:
  - Use GraalVM CE 22.3.1 instead of GraalVM CE 23.0 Dev. Refer to https://github.com/apache/shardingsphere/actions/runs/3998084528/jobs/6860258580 and https://github.com/graalvm/setup-graalvm/pull/31 .
  - Update OpenGauss JDBC to 3.1.0-og to match GraalVM reachability metadata. Refer to https://github.com/oracle/graalvm-reachability-metadata/pull/168 .
  - Update Commons DBCP2 to 2.9.0 to match GraalVM reachability metadata in the future nativeTest. Refer to https://github.com/oracle/graalvm-reachability-metadata/issues/202 .
  - Update native-maven-plugin to 0.9.19 to output richer Native Image build logs. Refer to https://graalvm.github.io/native-build-tools/latest/index.html#changelog .

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Review by linghengqian — COMMENTED** [review]
I'm not sure why a CI Job is timing out without an Error Log.

File: pom.xml [hunk 1]
```diff
@@ -94,14 +94,14 @@
         <lombok.version>1.18.20</lombok.version>
         
         <postgresql.version>42.4.1</postgresql.version>
-        <opengauss.version>3.0.0</opengauss.version>
+        <opengauss.version>3.1.0-og</opengauss.version>
```
Comment (TeslaCN) [inline pom.xml:97]: What's the different between `3.1.0` and `3.1.0-og`?

Comment (linghengqian) [reply pom.xml:97]: - I checked at https://github.com/oracle/graalvm-reachability-metadata/pull/168 and the difference is reflected in the package name. 

- `3.1.0-og` uses the same `org.opengauss.**` package name as `3.0.0`, while `3.1.0` uses the breakthrough `org.postgresql.**` as the package name, I can't find an explanation for this package name change from opengauss anywhere, it's ridiculous, I can only use `3.1.0-og`.

Comment (TeslaCN) [reply pom.xml:97]: Got it.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**